### ISSUE TITLE
Need to consider whether it is necessary to make allocation of space from blockdevs more efficient

### ISSUE THREAD

**Issue by mulkieran** [stratis-storage/stratisd#599]
The most obvious possible defect is that space is always allocated in order of the blockdevs addition to the pool.

**Comment by mulkieran** [comment]
See also #815. It is sort of wierd to allocate from the start but to have the blockdevs in a different order after setup. This could lead to some kind  of unnecessary fragmentation.

**Comment by agrover** [comment]
Blockdevs have a set order now. If some part of this issue still needs work @mulkieran please reopen, thanks

**Comment by mulkieran** [comment]
Yeah, we still allocate in order of blockdevs, including the ones that are already fully allocated. It's not an urgent problem, though.

**Comment by mulkieran** [comment]
There is plenty of solid work w/ filesystems and tracking allocations which would allow us to do a better job.
